### PR TITLE
Change to symmetric mining.

### PR DESCRIPTION
**Pull request by bovard**
Now your units will always request resources equally from all adjacent resource tiles. This should be a large fix for micro tree mining battles which heavily favored teams in a certain direction before.

### CODE REVIEW

File: src/Game/index.ts [hunk 1]
```diff
@@ -584,14 +599,125 @@ export class Game {
...
+        amountsReqs = amountsReqs.map(([amount, req]: [number, ResourceRequest]) => [amount - toFill, req]);
+        amountLeft -= toFill * amountsReqs.length
+        if (amountLeft < amountsReqs.length) {
+          amountLeft = 0;
+        }
+        amountsReqs = amountsReqs.filter(([amount, _]: [number, ResourceRequest]) => amount > 0);
+      }
+      // set the remaining amount to be the new cell total
+      const cell = this.map.getCellByPos(position);
+      cell.resource.amount = amountLeft;
```
Comment (bovard) [inline src/Game/index.ts:716]: If amountLeft is 0, should we set cell.resource to undefined?

Comment (StoneT2000) [reply src/Game/index.ts:716]: There's some separate part in the code that clears out 0 resource tiles

Comment (bovard) [reply src/Game/index.ts:716]: Sounds good

**Comment by StoneT2000** [comment]
lgtm

**Pull request merged**